duvet/src/config/schema/v0_4_0.rs

```diff
@@ -0,0 +1,356 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    annotation::AnnotationType,
+    config,
+    target::{Target, TargetPath},
+};
+use duvet_core::{diagnostic::IntoDiagnostic, path::Path, Result};
+use serde::Deserialize;
+use std::sync::Arc;
+
+use super::TemplatedString;
+
+#[derive(Debug, Deserialize)]
+#[serde(deny_unknown_fields)]
+#[cfg_attr(test, derive(schemars::JsonSchema))]
+pub struct Schema {
+    #[serde(default, rename = "source")]
+    pub sources: Arc<[Source]>,
+
+    #[serde(default, rename = "requirement")]
+    pub requirements: Arc<[Requirement]>,
+
+    #[serde(default)]
+    pub report: Arc<Report>,
+
+    #[serde(default, rename = "specification")]
+    pub specifications: Arc<[Specification]>,
+
+    #[serde(rename = "$schema")]
+    _schema: Option<Arc<str>>,
+}
+
+impl Schema {
+    pub fn load_sources(&self, sources: &mut Vec<config::Source>, root: &Path) -> Result {
+        for source in self.sources.iter() {
+            sources.push(config::Source {
+                // TODO add context to error
+                pattern: source.pattern.parse().into_diagnostic()?,
+                comment_style: (&source.comment_style).into(),
+                default_type: source.default_type.into(),
+                root: root.clone(),
+            });
+        }
+
+        Ok(())
+    }
+
+    pub fn load_requirements(
+        &self,
+        requirements: &mut Vec<config::Requirement>,
+        root: &Path,
+    ) -> Result {
+        // include several default paths
+        for pattern in [
+            ".duvet/requirements/**/*.toml",
+            ".duvet/todos/**/*.toml",
+            ".duvet/exceptions/**/*.toml",
```

Comment:
any docs this should be mentioned in so the user knows they don't need to add these?

Reply:
Yeah I will add that in the guide